### ISSUE TITLE
AmbiguousVariantSelectionException is thrown for android library projects

### ISSUE THREAD

**Issue by emmano** [jk1/Gradle-License-Report#119]
If an android library project defines a group as:

```
android {
    defaultConfig {
        group = "some.group.id"
    }
}
```
The library crashes throwing:

```
Caused by: org.gradle.internal.component.AmbiguousVariantSelectionException: More than one variant of project :lib matches the consumer attributes:
  - Configuration ':lib:debugApiElements' variant android-aidl:
      - Found artifactType 'android-aidl' but wasn't required.
      - Required com.android.build.api.attributes.BuildTypeAttr 'debug' and found compatible value 'debug'.
      - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
      - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' and found compatible value 'Aar'.
      - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
  - Configuration ':lib:debugApiElements' variant android-classes:
      - Found artifactType 'android-classes' but wasn't required.
      - Required com.android.build.api.attributes.BuildTypeAttr 'debug' and found compatible value 'debug'.
      - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
      - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' and found compatible value 'Aar'.
      - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
  - Configuration ':lib:debugApiElements' variant android-manifest:
      - Found artifactType 'android-manifest' but wasn't required.
      - Required com.android.build.api.attributes.BuildTypeAttr 'debug' and found compatible value 'debug'.
      - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
      - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' and found compatible value 'Aar'.
      - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
  - Configuration ':lib:debugApiElements' variant android-renderscript:
      - Found artifactType 'android-renderscript' but wasn't required.
      - Required com.android.build.api.attributes.BuildTypeAttr 'debug' and found compatible value 'debug'.
      - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
      - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' and found compatible value 'Aar'.
      - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
  - Configuration ':lib:debugApiElements' variant jar:
      - Found artifactType 'jar' but wasn't required.
      - Required com.android.build.api.attributes.BuildTypeAttr 'debug' and found compatible value 'debug'.
      - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
      - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' and found compatible value 'Aar'.
      - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
        at org.gradle.api.internal.artifacts.transform.AttributeMatchingVariantSelector.doSelect(AttributeMatchingVariantSelector.java:70)
        at org.gradle.api.internal.artifacts.transform.AttributeMatchingVariantSelector.select(AttributeMatchingVariantSelector.java:55)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.DefaultArtifactSet.select(DefaultArtifactSet.java:125)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.DefaultVisitedArtifactResults.select(DefaultVisitedArtifactResults.java:48)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.getSelectedArtifacts(DefaultLenientConfiguration.java:94)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.getFirstLevelNodes(DefaultLenientConfiguration.java:173)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.getFirstLevelModuleDependencies(DefaultLenientConfiguration.java:165)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.getFirstLevelModuleDependencies(DefaultLenientConfiguration.java:282)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultResolvedConfiguration.getFirstLevelModuleDependencies(DefaultResolvedConfiguration.java:67)
        at org.gradle.api.internal.artifacts.ivyservice.ErrorHandlingConfigurationResolver$ErrorHandlingResolvedConfiguration.getFirstLevelModuleDependencies(ErrorHandlingConfigurationResolver.java:280)
        at org.gradle.api.artifacts.ResolvedConfiguration$getFirstLevelModuleDependencies.call(Unknown Source)
        at com.github.jk1.license.reader.ConfigurationReader.read(ConfigurationReader.groovy:53)
        at com.github.jk1.license.reader.ConfigurationReader$read.call(Unknown Source)
        at com.github.jk1.license.reader.ProjectReader$_readConfigurationData_closure5.doCall(ProjectReader.groovy:82)
        at com.github.jk1.license.reader.ProjectReader.readConfigurationData(ProjectReader.groovy:80)
        at com.github.jk1.license.reader.ProjectReader$_read_closure1.doCall(ProjectReader.groovy:46)
        at com.github.jk1.license.reader.ProjectReader.read(ProjectReader.groovy:40)
        at com.github.jk1.license.reader.ProjectReader$read$0.call(Unknown Source)
        at com.github.jk1.license.ReportTask.generateReport(ReportTask.groovy:73)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        ... 30 more
```
The current solution is to remove the group declaration.

**Comment by emmano** [comment]
```
licenseReport {
    renderers = [new XmlReportRenderer(), new CsvReportRenderer(), new InventoryHtmlReportRenderer()]
    configurations = ALL
    projects = [project]
    excludeGroups = [rootProject.group]
}
```

**Comment by jk1** [comment]
I doubt that `configurations = ALL` is the best choice for android project. Android plugin creates lots of artificial configuration, most of which cannot be resolved at all. So I would specify something more relevant there.

**Comment by emmano** [comment]
What do you think is a proper configuration for an Android project? I have tried to specify one, but I go the same result.

**Comment by rviannaoliveira** [comment]
I have the same problem

**Comment by bandyguy** [comment]
@emmano Import your submodule projects like so and your issues should be resolved.
`project(path: ':submodule', configuration: 'default')`

**Comment by smidf** [comment]
The problem isn't in configuration, but in resolving classpath for task - https://github.com/jk1/Gradle-License-Report/blob/master/src/main/groovy/com/github/jk1/license/ReportTask.groovy#L36

**Comment by jk1** [comment]
@smidf please elaborate. This code is only used for an up-to-date check.

**Comment by smidf** [comment]
I tried debug this problem and running task (generateLicenseReport) finished with exception before running task action method - resolve classpath failed. I commented this method and task finished successfully. I think that it is similar problem as using `configurations = ALL` for Android project.

**Comment by jk1** [comment]
The fix is to be shipped in 1.11 and consists of the following improvements:
- Android projects now use a non-caching report task versions to avoid the issue described above
- Smarter configuration default has been provided for android projects - `releaseRuntimeClasspath`. I'm no Android expert, so any suggestions for a better default are welcome.
